### PR TITLE
Update DOM from TSJS-lib-generator

### PR DESCRIPTION
**Pull request by sandersn**
1. Update Navigator and other small updates.
2. Make ProgressEvent generic.
3. Make `window: Window & typeof globalThis`.
4. Add types:
    * CSS Overflow
    * CSS Masking
    * Web Authentication
    * WebGL 2

The big change is that `window` now includes globals in its type via `typeof globalThis`. This helps some codebases a lot, such as chrome-devtools-frontend.

### CODE REVIEW

**Comment by sandersn** [comment]
@typescript-bot user test this

**Comment by sandersn** [comment]
@typescript-bot test this

**Comment by sandersn** [comment]
@typescript-bot run dt

**Comment by sandersn** [comment]
User tests look good; no surprises there. Chrome-devtools-frontend especially benefits.

**Comment by sandersn** [comment]
From RWC:

1. We lose related spans on members declared as part of Window and declared globally:

```ts
window.navigate
       ~~~~~~~~
       'navigate' not found on 'Window & typeof globalThis'. Did you mean 'navigator'?
```

But it no longer has a related span "navigator is declared here", because navigator now has two declarations.

2. Casting `window as MyMadeupWindowSubtype` will not work until `MyMadeupWindowSubtype` is revised to include properties from globalThis, or the cast is doubled to cast through `unknown`.

3. VS Code walks a parent chain of windows, but starts the loop with `let w = window`. It now needs to be `let w: Window = window`.

I don't think any of these breaks are bad.

**Comment by sandersn** [comment]
DT looks fine, just a couple of `$ExpectType` comments need to be updated.

File: src/lib/dom.generated.d.ts [hunk 1]
```diff
@@ -4327,59 +4444,59 @@ interface DocumentEventMap extends GlobalEventHandlersEventMap, DocumentAndEleme
...
-     * Deprecated. Sets or retrieves a value that indicates the background color behind the object.
+    /**
+     * Deprecated. Sets or retrieves a value that indicates the background color behind the object.
      */
     /** @deprecated */
     bgColor: string;
-    /**
-     * Specifies the beginning and end of the document body.
+    /**
+     * Specifies the beginning and end of the document body.
```
Comment (mihailik) [inline src/lib/dom.generated.d.ts]: 👆 what is all this, LF/CRLF variation?

Looks set to create trouble down the line.

Comment (sandersn) [reply src/lib/dom.generated.d.ts]: Looks like comments.json (1) still has explicit `\r\n` and (2) something changed recently to stop converting them to `\n`.

The right fix is to explicitly change comments.json to use `\n`.

Comment (mihailik) [reply src/lib/dom.generated.d.ts]: Good idea!

People may have weird git configs, check in and then CRLF noise could obscure real changes for PR reviews.

**Pull request merged**